File: exampleCode/cpp_inheritance/assert_test_test/testing_code/Person/wrong/Person.cpp
//#inlcude "Person.h"
#include <string>
#include <iostream>

using namespace std;
/*
class Person: public ProtoPerson
{
 public:


Person()
{
	init("John","Doe",0);

}

Person(string fname,string lname)
{
	init(fname,lname,0);

}

Person(string fname,string lname,int age)
{
	init(fname,lname,age);

}


void setLastName(string s)
{
  setFirstName(s);
}

  

};
*/


#include "Person.h"


void 
Person::setLastName(string s)
{
  setFirstName(s);
}




Person::Person()
{
	init("John","Doe",0);

}

Person::Person(string fname,string lname)
{
	init(fname,lname,0);

}

Person::Person(string fname,string lname,int age)
{
	init(fname,lname,age);

}




Person operator +(const Person& A, const Person& B)
{
  Person C(A.getFirstName(),B.getLastName(),0);
	return C;
}

File: exampleCode/python/assert_test/example_solution/Person.py
# Person.py
#
# Michael K. Bradshaw
#
# June 2014

class Person(object):
    
    # fname - first name
    # lname - last name
    
    def __init__(self,fname,lname):
        self.fname = fname
        self.lname = lname

    def getFirstName(self):
        return self.fname

    def getLastName(self):
        return self.lname

    def setFirstName(self,fname):
        self.fname=fname

    def setLastName(self,lname):
        self.lname=lname

    def __str__(self):
        s = self.fname +" "+ self.lname
        return s






